UnitTests/UnitTestController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using GetandTake.Controller;
+using GetandTake.Models.DTOs.ListDTO;
+using GetandTake.Services.Abstract;
+using Moq;
+
+namespace UnitTests;
+
+public class UnitTestController
+{
+    [Fact]
+    public Task ProductList_ShouldExecuteSuccessullyAndReturnProducts()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProductsAsync();
+        productService.Setup(product => product.GetAllAsync())
+            .Returns(productList);
+        var productController = new ProductController(productService.Object);
+
+        //act
+        var productResult = productController.ProductList();
+
+        //assert
+        Assert.NotNull(productResult);
+        Assert.Equal(productList.Result.Count, productResult.Result.Count);
+        Assert.Equal(productList.ToString(), productResult.ToString());
+        return Task.CompletedTask;
+    }
+
+    [Fact]
+    public Task Product_ShouldExecuteSuccessullyAndReturnProduct()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProductsAsync().Result;
+        productService.Setup(product => product.GetByIdAsync(2).Result)
+            .Returns(productList[1]);
+        var productController = new ProductController(productService.Object);
+
+        //act
+        var productResult = productController.GetProductByIdAsync(2);
+
+        //assert
+        Assert.NotNull(productResult);
+        Assert.Equal(productList[1].ProductID, productResult.Result.ProductID);
+        Assert.True(productList[1].ProductID == productResult.Result.ProductID);
```

Comment:
No need to use `Assert.True` here since you have already compared values of this item in row above